hypothesis-extra/hypothesis-pytest/src/hypothesispytest/__init__.py

```diff
@@ -19,17 +19,20 @@
 
 class StoringReporter(object):
 
-    def __init__(self):
+    def __init__(self, config):
+        self.config = config
         self.results = []
 
     def __call__(self, msg):
+        if self.config.getoption('capture', 'fd') == 'no':
+            print(msg)
```

Comment:
Hmm. That's troubling. Are you sure the hook wrapper is being executed way before any tests? Because if it is this (i.e. my code) is probably only doing the right thing by coincidence anyway.